assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { render, findByLabelText, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * Internal dependencies
+ */
+import FrontendBlock from '../frontend';
+
+describe( 'FrontendBlock', () => {
+	let validationData = {
+		hasValidationErrors: false,
+		getValidationError: jest.fn(),
+		clearValidationError: jest.fn(),
+		hideValidationError: jest.fn(),
+		setValidationErrors: jest.fn(),
+	};
+	beforeEach( () => {
+		validationData = {
+			hasValidationErrors: false,
+			getValidationError: jest.fn(),
+			clearValidationError: jest.fn(),
+			hideValidationError: jest.fn(),
+			setValidationErrors: jest.fn(),
+		};
+	} );
+
+	it( "Renders a checkbox if the checkbox prop is passed and doesn't if it's not.", async () => {
```

Comment:
What do you think about splitting this out into two tests? I found it a bit confusing reading this

Reply:
Yeah I agree, done!